File: android_smart_glasses/main/app/src/main/java/com/wearableintelligencesystem/androidsmartglasses/comms/gatt/DeviceConnectionState.java
package com.wearableintelligencesystem.androidsmartglasses.comms.gatt;

import android.bluetooth.BluetoothDevice;

public abstract class DeviceConnectionState {

    private int state;

    private DeviceConnectionState(int state) {
        this.state = state;
    }

    public int getState() {
        return state;
    }

    public static class Connected extends DeviceConnectionState {

        private BluetoothDevice device;

        public Connected(BluetoothDevice device) {
            super(2);
            this.device = device;
        }

        public BluetoothDevice getDevice() {
            return device;
        }
    }

    public static class Connecting extends DeviceConnectionState {
        public Connecting() {
            super(1);
        }
    }

    public static class Disconnected extends DeviceConnectionState {
        public Disconnected() {
            super(0);
        }
    }
}

File: android_smart_phone/main/app/src/main/java/com/smartglassesmanager/androidsmartphone/smartglassescommunicators/gatt/DeviceConnectionState.kt

package com.smartglassesmanager.androidsmartphone.smartglassescommunicators.gatt

import android.bluetooth.BluetoothDevice

sealed class DeviceConnectionState( val state:Int ) {
    class Connected(val device: BluetoothDevice) : DeviceConnectionState(2)
    object Connecting:DeviceConnectionState(1)
    object Disconnected : DeviceConnectionState(0)
}

